design/database.md

```diff
@@ -0,0 +1,48 @@
+# Database Design
+
+## Table: pipeline\_groups
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string | `NOT NULL`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: pipelines
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| name | string | `NOT NULL`
+| pipeline\_group\_id | belongs\_to | `NOT NULL`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: pipelines\_materials
+| Field name | Data type | Extra
+| --- | --- | ---
+| pipeline\_id | integer | `compound_key`
+| material\_id | integer | `compound_key`
+
+## Table: materials
+| Field name | Data type | Extra
+| --- | --- | ---
+| id | integer | `primary_key`
+| type | string | `enum: {git|file|...}`
+| url | string | `{git://...|file://source_pipeline/release.tar.gz}`
+| inserted\_at | datetime |
+| updated\_at | datetime |
+
+## Table: environment\_variables
```

Comment:
This might mess with the keep it simple, so I'm just going to suggest it, but we might want to change it.
 
Can we subdivide environment variables into groups? Is that a bad idea?

Reply:
To explain a bit more. I sometimes wish that I could create groups of env vars that I could enable and disable depending on the material branch.

Reply:
Won't that couple the envvars to the scm implementation though? I think composition of envvars is an important feature but I'm not sure what the best way to go about it is yet - might be good to try a few ideas out

Reply:
👍